File: 0899-orderly-queue/0899-orderly-queue.cpp
class Solution {
public:
    string orderlyQueue(string s, int k) {
        
        if(k >1){
            sort(s.begin() , s.end());
            return s; 
        }
        int n = s.size();
        string ans = s; 
        s +=s ; 
        for(int i=0; i< n ; i++){
            if(s.substr(i,n) < ans )
                ans = s.substr(i,n);
        }
        return ans; 
        
        
    }
};

File: 0899-orderly-queue/0899-orderly-queue.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var orderlyQueue = function(s, k) {
    
    if(k==1){
        let first = s.split('').sort()[0];
        let ans = [] ;
        for(let i=0; i< s.length ;i++){
            if(s[i]==first){
                ans.push( s.slice(i) +s.slice(0,i));
            }
        }
        return ans.sort()[0];
    }
    else {
        return s.split('').sort().join('') ; 
    }   
};